lib/events/s3sessions/s3stream.go

```diff
@@ -89,16 +91,30 @@ func (h *Handler) UploadPart(ctx context.Context, upload events.StreamUpload, pa
 		"key":     uploadKey,
 	})
 
+	// Calculate the content MD5 hash to be included in the request. This is required for S3 buckets with Object Lock enabled.
+	hash := md5.New()
+	if _, err := io.Copy(hash, partBody); err != nil {
+		return nil, trace.Wrap(err, "failed to calculate content MD5 hash")
+	}
+	md5sum := base64.StdEncoding.EncodeToString(hash.Sum(nil))
```

Comment:
I'm a bit confused about this based on their docs: https://github.com/aws/aws-sdk-go-v2/blob/c8a827488ef90991f42e319fb5da4b2961d89518/feature/s3/manager/upload.go#L190-L191

```
// ContentMD5 for multipart upload is explicitly ignored for multipart upload,
// and its value is suppressed.
```